src/pal/src/file/file.cpp

```diff
@@ -1250,8 +1287,17 @@ DeleteFileW(
       lpFileName?lpFileName:W16_NULLSTRING);
 
     pThread = InternalGetCurrentThread();
-    size = WideCharToMultiByte( CP_ACP, 0, lpFileName, -1, name, MAX_LONGPATH,
+    
+    if (lpFileName != NULL)
+    {
+        length = (PAL_wcslen(lpFileName)+1) * sizeof(WCHAR);
```

Comment:
Actually, after seeing more of the code, I can see it would be even better to extract the whole conversion into a separate function, something like 
```C++
int ConvertPathToAcp(LPCWSTR srcPath, PathCharString* dstPath)
{
    int length = (PAL_wcslen(srcPath)+1) * MaxWCharToAcpLengthRatio;
    dstPath->OpenStringBuffer(length);
    int size = WideCharToMultiByte( CP_ACP, 0, srcPath, -1, destPath, length, NULL, NULL );
    dstPath->CloseBuffer(size);
    return size;
}
```